africanlii/views/documents.py

```diff
@@ -50,11 +52,18 @@ def render_to_response(self, context, **response_kwargs):
 
 class DocumentSourcePDFView(DocumentSourceView):
     def render_to_response(self, context, **response_kwargs):
-        if hasattr(self.object, "source_file") and self.object.source_file.file:
-            if self.object.source_file.file.name.endswith(".docx"):
-                temp_dir, filename = convert_docx_to_pdf(self.object.source_file.file)
-                file = open(f"{temp_dir.name}/{filename}", "rb")
-            else:
-                file = self.object.source_file.file.open()
-            return view_attachment(file)
-        raise Http404
+        infile = self.object.source_file.file
```

Comment:
if `source_file` exists, it should always have a file, so you don't need to check for that.